app/models/file.js

```diff
@@ -1,46 +1,45 @@
-import Model, { belongsTo, attr } from '@ember-data/model';
-import { alias } from '@ember/object/computed';
-import { computed } from '@ember/object';
+import Model, { attr, belongsTo } from '@ember-data/model';
+import { deprecate } from '@ember/application/deprecations';
 import sanitize from 'sanitize-filename';
 
-// TODO: octane-refactor
-/* eslint-disable ember/no-get */
-// eslint-disable-next-line ember/no-classic-classes
-export default Model.extend({
-  piece: belongsTo('piece'),
-
-  filename: attr('string'),
-  filenameWithoutExtension: computed('filename', 'extension', {
-    get() {
-      const ext = this.get('extension');
-      // eslint-disable-next-line no-useless-escape
-      const regex = new RegExp(`\.${ext}$`);
-      return this.get('filename').replace(regex, '');
-    },
-    // eslint-disable-next-line no-unused-vars
-    set(key, value) {
-      const filename = `${value}.${this.get('extension')}`;
-      this.set('filename', filename);
-      return value;
-    },
-  }),
-
-  format: attr('string'),
-  size: attr('number'),
-  extension: attr('string'),
-  created: attr('datetime'),
-  contentType: attr('string'),
-
-  downloadFilename: computed('filename', function() {
-    return sanitize(this.get('filename'), { // file-system-safe
+export default class File extends Model {
+  @attr('string') filename;
+  @attr('string') format;
+  @attr('number') size;
+  @attr('string') extension;
+  @attr('datetime') created;
+  @attr('string') contentType;
+
+  @belongsTo('piece') piece;
+
+  get name() {
+    deprecate(`Attribute 'name' on 'file' model is deprecated. Use 'filename' instead.`);
+    return this.filename;
+  }
+
+  get downloadFilename() {
+    return sanitize(this.filename, {
       replacement: '_',
     });
-  }),
-  downloadLink: computed('id', function() {
-    return `/files/${this.get('id')}/download`;
-  }),
-  namedDownloadLink: computed('downloadFilename', 'downloadLink', function() {
-    return `${this.downloadLink}?name=${encodeURIComponent(this.downloadFilename)}`; // url-safe
-  }),
-  name: alias('filename'), // Compatibility. Use of 'name' should be refactored out.
-});
+  }
+
+  get downloadLink() {
+    return `/files/${this.id}/download`;
+  }
+
+  get namedDownloadLink() {
+    return `${this.downloadLink}?name=${encodeURIComponent(this.downloadFilename)}`;
+  }
+
+  get filenameWithoutExtension() {
+    const regex = new RegExp(`.${this.extension}$`);
+    return this.filename.replace(regex, '');
+  }
+
+  get canPreview() {
```

Comment:
Ik zie dat je deze test naar het model refactored hebt, maar hiervan zou ik durven argumenteren dat het toch eerder een eigenschap van de view-component is dan van het model (hoewel dit in de praktijk nog niet voorvalt, zou je je kunnen inbeelden dat 1 component enkel pdf's ondersteunt, maar een andere bv. ook word-files)

Reply:
Ik zat er vooral mee verveeld dat die code nu op 2 plaatsen gedupliceerd was en de relation die erin opgehaald wordt niet ge-await werd.

Je hebt gelijk dat er in de toekomst meer viewers kunnen zijn. Ik begin daarom meer te denken richting verantwoordelijkheid van de view-component zelf ipv de route/controller errond. Aangezien we die `PdfView`-component waarschijnlijk niet afhankelijk willen maken van ons file data model, misschien een wrapping component `DocumentView` waaraan we `@file` meegegeven en die dan zelf beslist met welke viewer hij het document toont (in dit geval dus PDF of niets, maar kan uitgebreid worden met andere viewers in de toekomst). Wat denk je?

Reply:
Akkoord voor `DocumentView`. Zo was het oorspronkelijk trouwens ook voorzien :-) , maar omdat dat voor de eerste tijd een onnodig niveau van indirectie was, is dat er terug uit gegaan ...